src/spi.rs

```diff
@@ -233,6 +233,10 @@ pub mod config {
         pub duplex: Duplex,
         pub bit_order: BitOrder,
         pub cs_active_high: bool,
+        ///< Amount of SPI bit-cycles the cs should be activated before the transmission (0-16). This only works on half-duplex transactions.
```

Comment:
If you specify more than 16 'bit cycles', how does the api react? does it take 16 than? If not sure can you test it and report back?

Reply:
It doesn't change anything given my test (0/1/16/1000), at least in FullDuplex Mode. It behaves like if the value was the highest available. 
Further testing on other boards and configs could be interesting.

Reply:
I am doing some further testing at home on another oscilloscope and another ESP32 (its still a NodeMcu but instead of having written ESP32S on top its writtern ESP32-WROOM-32E). It behaves the same as the other one, but for scientific purpose I will redo all the tests.
Just to note that DMA doesn't work on Half-Duplex (I think this is normal and documented) (which I was activating using Fullduplex in my previous tests and no error was shown about it). I will post the results once I finish.

Reply:
thanks for further looking into it. If you want to check out the C driver path. We are using the somewhat stable driver/spi C api, though it itself goes down a bit deeper and is somewhat splitted.
1. In C land they have the high level drivers for General Purpose (gpspi) and one for communicating with the internal Flash memory (sdspi). This is what lives in https://github.com/espressif/esp-idf/blob/master/components/driver/spi/include/driver/spi_master.h and neighborhood.
2. This drivers are written against an C hal living in https://github.com/espressif/esp-idf/blob/master/components/hal/include/hal/spi_hal.h
3. The hal itself acts on a "low level" api that is somewhat simmilar accros all chips. This is the abstraction over the register level. Here we can also find more infos about the delays itself. For example if we look at the codepath for an esp32 can be found here. https://github.com/espressif/esp-idf/blob/master/components/hal/esp32c3/include/hal/spi_ll.h#L795 . There is an interesting command that suggest the delay should not work in Full-Duplex if command and address features are utilized, but we are not using them. So by that comment it should work in Full-Duplex. Though your test seems to suggest that is not really working fully just "partially"?